src/generator/default/dbmodel.php

```diff
@@ -37,10 +37,38 @@
  */
 abstract class <?= $model->getClassName() ?> extends \yii\db\ActiveRecord
 {
+<?php if (count($model->virtualAttributes())):?>
+    protected $virtualAttributes = ['<?=implode("', '", array_map(function ($attr) {
+    return $attr->columnName;
+}, $model->virtualAttributes()))?>'];
+
+<?php foreach ($model->virtualAttributes() as $attribute): ?>
+    /**
+     * @var <?=$attribute->phpType.PHP_EOL?>
+    */
+    public $<?= $attribute->columnName ?>;
+
+<?php endforeach; ?>
+<?php endif;?>
     public static function tableName()
     {
         return <?= var_export($model->getTableAlias()) ?>;
     }
+<?php if (count($model->virtualAttributes())):?>
+
+    public function attributes()
+    {
+        return array_merge(parent::attributes(), $this->virtualAttributes);
+    }
+
+    public function afterFind()
+    {
+        parent::afterFind();
+        foreach ($this->virtualAttributes as $attr) {
+            $this->$attr = $this->getAttribute($attr);
+        }
```

Comment:
what is this code supposed to do? `$this->getAttribute()` would return `null` for all attributes that do not come from the database.

Reply:
Oh, it is hard to remember the reason now...  but seems it was. Will recheck it